### PR TITLE
More resilient qui-disk-space widget

### PR DESCRIPTION
**Pull request by marmarta**
Wrap pool access in intermediary object to catch all possible exceptions. Cleanup for more readability.

fixes QubesOS/qubes-issues#8187

### CODE REVIEW

**Comment by marmarta** [comment]
PipelineRetryFailed

File: qui/tray/disk_space.py [hunk 1]
```diff
@@ -160,31 +195,29 @@ def __populate_pools(self):
...
+                continue
+            self.total_size += wrapped_pool.size
+            self.used_size += wrapped_pool.usage
+
+            if wrapped_pool.usage_perc >= URGENT_WARN_LEVEL:
+                self.warning_message.append(
+                    _("\n{:.1%} space left in pool {}").format(
+                        1-wrapped_pool.usage_perc, wrapped_pool.name))
+
+            if wrapped_pool.metadata_usage >= URGENT_WARN_LEVEL:
```
Comment (marmarek) [inline qui/tray/disk_space.py:215]: metadata_perc?

Currently it fails this way for pools without metadata usage:
```
May 23 10:05:41 dom0 widget-wrapper[3024]: Traceback (most recent call last):
May 23 10:05:41 dom0 widget-wrapper[3024]:   File "/usr/bin/qui-disk-space", line 33, in <module>
May 23 10:05:41 dom0 widget-wrapper[3024]:     sys.exit(load_entry_point('qui==0.1', 'gui_scripts', 'qui-disk-space')())
May 23 10:05:41 dom0 widget-wrapper[3024]:              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
May 23 10:05:41 dom0 widget-wrapper[3024]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 506, in main
May 23 10:05:41 dom0 widget-wrapper[3024]:     app = DiskSpace()
May 23 10:05:41 dom0 widget-wrapper[3024]:           ^^^^^^^^^^^
May 23 10:05:41 dom0 widget-wrapper[3024]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 355, in __init__
May 23 10:05:41 dom0 widget-wrapper[3024]:     self.refresh_icon()
May 23 10:05:41 dom0 widget-wrapper[3024]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 362, in refresh_icon
May 23 10:05:41 dom0 widget-wrapper[3024]:     pool_data = PoolUsageData(self.qubes_app)
May 23 10:05:41 dom0 widget-wrapper[3024]:                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
May 23 10:05:41 dom0 widget-wrapper[3024]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 190, in __init__
May 23 10:05:41 dom0 widget-wrapper[3024]:     self.__populate_pools()
May 23 10:05:41 dom0 widget-wrapper[3024]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 215, in __populate_pools
May 23 10:05:41 dom0 widget-wrapper[3024]:     if wrapped_pool.metadata_usage >= URGENT_WARN_LEVEL:
May 23 10:05:41 dom0 widget-wrapper[3024]:        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
May 23 10:05:41 dom0 widget-wrapper[3024]: TypeError: '>=' not supported between instances of 'NoneType' and 'float'
```

Comment (marmarta) [reply qui/tray/disk_space.py]: you're right, pushed a fix

**Comment by marmarek** [comment]
Now I've got this:
```
May 24 13:38:33 dom0 widget-wrapper[2660]: Traceback (most recent call last):
May 24 13:38:33 dom0 widget-wrapper[2660]:   File "/usr/bin/qui-disk-space", line 33, in <module>
May 24 13:38:33 dom0 widget-wrapper[2660]:     sys.exit(load_entry_point('qui==0.1', 'gui_scripts', 'qui-disk-space')())
May 24 13:38:33 dom0 widget-wrapper[2660]:              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
May 24 13:38:33 dom0 widget-wrapper[2660]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 506, in main
May 24 13:38:33 dom0 widget-wrapper[2660]:     app = DiskSpace()
May 24 13:38:33 dom0 widget-wrapper[2660]:           ^^^^^^^^^^^
May 24 13:38:33 dom0 widget-wrapper[2660]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 355, in __init__
May 24 13:38:33 dom0 widget-wrapper[2660]:     self.refresh_icon()
May 24 13:38:33 dom0 widget-wrapper[2660]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 362, in refresh_icon
May 24 13:38:33 dom0 widget-wrapper[2660]:     pool_data = PoolUsageData(self.qubes_app)
May 24 13:38:33 dom0 widget-wrapper[2660]:                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
May 24 13:38:33 dom0 widget-wrapper[2660]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 190, in __init__
May 24 13:38:33 dom0 widget-wrapper[2660]:     self.__populate_pools()
May 24 13:38:33 dom0 widget-wrapper[2660]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 219, in __populate_pools
May 24 13:38:33 dom0 widget-wrapper[2660]:     "Current usage: {.1%}").format(
May 24 13:38:33 dom0 widget-wrapper[2660]:                             ^^^^^^^
May 24 13:38:33 dom0 widget-wrapper[2660]: AttributeError: 'float' object has no attribute '1%'
```

**Comment by marmarek** [comment]
```
May 26 00:57:28 dom0 widget-wrapper[2737]: Traceback (most recent call last):
May 26 00:57:28 dom0 widget-wrapper[2737]:   File "/usr/bin/qui-disk-space", line 33, in <module>
May 26 00:57:28 dom0 widget-wrapper[2737]:     sys.exit(load_entry_point('qui==0.1', 'gui_scripts', 'qui-disk-space')())
May 26 00:57:28 dom0 widget-wrapper[2737]:              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
May 26 00:57:28 dom0 widget-wrapper[2737]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 506, in main
May 26 00:57:28 dom0 widget-wrapper[2737]:     app = DiskSpace()
May 26 00:57:28 dom0 widget-wrapper[2737]:           ^^^^^^^^^^^
May 26 00:57:28 dom0 widget-wrapper[2737]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 355, in __init__
May 26 00:57:28 dom0 widget-wrapper[2737]:     self.refresh_icon()
May 26 00:57:28 dom0 widget-wrapper[2737]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 362, in refresh_icon
May 26 00:57:28 dom0 widget-wrapper[2737]:     pool_data = PoolUsageData(self.qubes_app)
May 26 00:57:28 dom0 widget-wrapper[2737]:                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
May 26 00:57:28 dom0 widget-wrapper[2737]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 190, in __init__
May 26 00:57:28 dom0 widget-wrapper[2737]:     self.__populate_pools()
May 26 00:57:28 dom0 widget-wrapper[2737]:   File "/usr/lib/python3.11/site-packages/qui/tray/disk_space.py", line 219, in __populate_pools
May 26 00:57:28 dom0 widget-wrapper[2737]:     "Current usage: {0:.0%}").format(
May 26 00:57:28 dom0 widget-wrapper[2737]:                               ^^^^^^^
May 26 00:57:28 dom0 widget-wrapper[2737]: ValueError: cannot switch from automatic field numbering to manual field specification
```

**Comment by marmarek** [comment]
It doesn't crash anymore, but clearly something is still wrong:
https://openqa.qubes-os.org/tests/74488#step/qui_widgets_disk_space/4

**Review by marmarek — APPROVED** [review]

**Pull request merged**